SharedPackages/BrowserServicesKit/Sources/Subscription/Managers/SubscriptionManagerV2.swift

```diff
@@ -482,7 +474,9 @@ public final class DefaultSubscriptionManagerV2: SubscriptionManagerV2 {
 
     public func signOut(notifyUI: Bool) async {
         Logger.subscription.log("SignOut: Removing all traces of the subscription and account")
-        try? await oAuthClient.logout()
+        Task { // Not waiting for an answer
+            try? await oAuthClient.logout()
+        }
```

Comment:
I think this is counterintuitive and we may just be begging for problems. When wrapping it with `Task` you make the logout asynchronous and out of sync with the notification being posted. I imagine somewhere in the code we may receive the `.accountDidSignOut` notification that will execute in a brief state when user will still be signed in.

Reply:
Agree, I made the `oAuthClient.logout()` sync again and moved the async part inside oAuthClient, now only the API logout call is asynchronous, and that still solves the issue.